extensions/fairifier-rdf-extension/dtls/fairifier/PostFairDataToFairDataPoint.java

```diff
@@ -319,8 +231,42 @@ public void doPost(HttpServletRequest req, HttpServletResponse res)
             writer.value(distributionPost);
             writer.endObject();
 
-        } catch (Exception ex) {
+        } catch (
+
+        Exception ex) {
             respondException(res, ex);
         }
     }
+
+    private <T extends Metadata> T getMetadata(Class<T> type, JSONObject args, String uri)
+            throws JSONException, InstantiationException, IllegalAccessException {
+        T metadata = type.newInstance();
+        metadata.setVersion(f.createLiteral(args.getString("http://purl.org/dc/terms/hasVersion")));
+        metadata.setUri(f.createIRI(uri));
+        Date date = new Date();
+        metadata.setIssued(f.createLiteral(date));
+        metadata.setModified(f.createLiteral(date));
+        metadata.setTitle(f.createLiteral(args.getString("http://purl.org/dc/terms/title")));
+        if (args.has("http://purl.org/dc/terms/rights")) {
+            metadata.setRights(f.createIRI(args.getString("http://purl.org/dc/terms/rights")));
+        }
+        if (!metadata.getClass().equals(DistributionMetadata.class)) {
+            Agent agent = new Agent();
+            agent.setUri(f.createIRI(args.getJSONObject("http://purl.org/dc/terms/publisher").getString("url")));
+            agent.setName(f.createLiteral(args.getJSONObject("http://purl.org/dc/terms/publisher").getString("url")));
+            metadata.setPublisher(agent);
+        }
+        return metadata;
+    }
+
+    private String pushMetadataToFdp(String uuid, Metadata metadata, String metadataType, String baseUri)
+            throws MetadataException, JSONException, IOException {
+
+        String metadataLayerStr = MetadataUtils.getString(metadata, RDFFormat.TURTLE, false)
+                .replaceAll("\\<" + metadata.getUri().toString() + "\\>", "<>");
+        String metadataId = baseUri + "/" + metadataType + "?id=" + metadata.getTitle().getLabel().replace(" ", "_")
+                + "_" + metadata.getVersion().getLabel().replace(" ", "_") + "_" + uuid;
+
+        return IOUtils.toString(HttpUtils.post(metadataId, metadataLayerStr, "text/turtle").getContent(), "UTF-8");
```

Comment:
`HttpUtils.post()` probably returns a response like object where we can get the status code. This might help in retrieving useful (error) messages for the front-end user.